src/app/zap-templates/zcl/data-model/chip/device-energy-management-cluster.xml

```diff
@@ -0,0 +1,192 @@
+    <item name="HomeWaterPump" value="0x0B"/>
+    <item name="IrrigationWaterPump" value="0x0C"/>
+    <item name="PoolPump" value="0x0D"/>
+    <item name="Other" value="0xFF"/>
+  </enum>
+  <enum name="ESAStateEnum" type="enum8" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item name="Offline" value="0x00"/>
+    <item name="Online" value="0x01"/>
+    <item name="Fault" value="0x02"/>
+    <item name="UserOptOut" value="0x03"/>
+    <item name="PowerAdjustActive" value="0x04"/>
+    <item name="Paused" value="0x05"/>
+  </enum>
+  <enum name="CauseEnum" type="enum8" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item name="NormalCompletion" value="0x00"/>
+    <item name="Offline" value="0x01"/>
+    <item name="Fault" value="0x02"/>
+    <item name="UserOptOut" value="0x03"/>
+  </enum>
+  <struct name="CostStruct" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item fieldId="0" name="CostType" type="CostTypeEnum"/>
+    <item fieldId="1" name="Value" type="int32s"/>
+    <item fieldId="2" name="DecimalPoints" type="int8u"/>
+    <item fieldId="3" name="Currency" type="int16u" max="999" optional="true"/>
+  </struct>
+  <struct name="PowerAdjustStruct" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item fieldId="0" name="MinPower" type="int64s"/>
+    <item fieldId="1" name="MaxPower" type="int64s"/>
+    <item fieldId="2" name="MinDuration" type="elapsed_s"/>
+    <item fieldId="3" name="MaxDuration" type="elapsed_s"/>
+  </struct>
+  <struct name="ForecastStruct" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item fieldId="0" name="ForecastId" type="int16u"/>
+    <item fieldId="1" name="ActiveSlotNumber" type="int16u" max="0xFFFE" isNullable="true"/>
+    <item fieldId="2" name="StartTime" type="epoch_s"/>
+    <item fieldId="3" name="EndTime" type="epoch_s"/>
+    <item fieldId="4" name="EarliestStartTime" type="epoch_s" max="0xFFFFFFFE" isNullable="true" optional="true"/>
+    <item fieldId="5" name="LatestEndTime" type="epoch_s" optional="true"/>
+    <item fieldId="6" name="IsPauseable" type="boolean"/>
+    <item fieldId="7" name="Slots" array="true" type="SlotStruct"/>
+  </struct>
+  <struct name="SlotStruct" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item fieldId="0" name="MinDuration" type="elapsed_s"/>
+    <item fieldId="1" name="MaxDuration" type="elapsed_s"/>
+    <item fieldId="2" name="DefaultDuration" type="elapsed_s"/>
+    <item fieldId="3" name="ElapsedSlotTime" type="elapsed_s"/>
+    <item fieldId="4" name="RemainingSlotTime" type="elapsed_s"/>
+    <item fieldId="5" name="SlotIsPauseable" type="boolean"/>
+    <item fieldId="6" name="MinPauseDuration" type="elapsed_s"/>
+    <item fieldId="7" name="MaxPauseDuration" type="elapsed_s"/>
+    <item fieldId="8" name="ManufacturerESAState" type="int16u" optional="true"/>
+    <item fieldId="9" name="NominalPower" type="int64s" optional="true"/>
```

Comment:
The types here don't match to the spec PR you gave - are power-mW, energy-mWh and amerage-mA defined types that should be used here?

Reply:
Correct - however at the current time the ZAP doesn't recognise these new types, which are underlying int64s.

I'm not sure if this needs a change in ZAP but there are multiple areas which these types need to go (e.g. searching for epoch_s finds things in:
examples/common/pigweed/rpc_services/Attributes.h
examples/common/pigweed/protos/attributes_service.proto
scripts/py_matter_idl/matter_idl/generators/type_definitions.py
scripts/py_matter_yamltests/matter_yamltests/constraints.py
scripts/tools/zap/tests/outputs/all-clusters-app/app-templates/endpoint_config.h
src/app/zap-templates/common/override.js

There is a similar issue filed for EVSE.xml already (#30665).

The proposal is to fix this later, but will need some help on zap tool since I'm getting these errors:
    (SELECT
      DATA_TYPE_ID
     FROM
      DATA_TYPE
     WHERE
      DATA_TYPE.PACKAGE_REF IN (2)
      AND DATA_TYPE.NAME = ?)), values: ChargingTargetStruct,153,ChargingTargetStruct,ChargingTargetStruct,153,AddedEnergy,2,false,false,0,,false,false,true,false,,energy_mWh, Error: SQLITE_CONSTRAINT: NOT NULL constraint failed: STRUCT_ITEM.DATA_TYPE_REF

(after doing a rm -rf ~/.zap)
:(

Reply:
It looks like I can get energy_mwh to work, but not energy_mWh etc due to some toLowerCase() things in ZAP.
If we can merge this PR (which is blocking other work) then when #30665 is resolved this will get resolved.